[18.01] Hack paster args to not duplicate pid file stuff for RUN_ALL.

lparsons opened the pull request:
Starting galaxy with RUN_ALL and using paste would attempt to use `galaxy.pid` and `galaxy.log` for every server process due to the duplicated args in `paster_args`. This was discussed on gitter with @natefoo and @jmchilton .

lparsons committed:
Hack paster args to not duplicate pid file stuff for RUN_ALL.

nsoranzo commented:
I think this would make `run.sh` not respect the environment variable `GALAXY_LOG`, and similarly for `run_tool_shed.sh` and `run_reports.sh`.

lparsons commented:
@nsoranzo Does this mean no plans to fix this until 18.09? Unfortunately, it renders `run.sh` totally broken for paste instances, so without a fix, I'll have to maintain a patch (or give up on paste and move to uWSGI which I'll have to do soon enough).

nsoranzo commented:
@lparsons No, I just think that's an incomplete fix, the options need to be added somehow when GALAXY_RUN_ALL is not set.

lparsons commented:
Ah, ok, I wasn't sure what the milestone meant... I'm sure it's incomplete. Will see if I can take a look sometime soon. This is currently my blocking issue for updating, though I may just patch it and move on... ;)

--
Lance Parsons - Scientific Programmer
Carl C. Icahn Laboratory - Room 136
Lewis-Sigler Institute for Integrative Genomics
Princeton University




On Wed, May 30, 2018 at 4:08 PM -0400, "Nicola Soranzo" <notifications@github.com<mailto:notifications@github.com>> wrote:


@lparsons<https://github.com/lparsons> No, I just think that's an incomplete fix, the options need to be added somehow when GALAXY_RUN_ALL is not set.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/galaxyproject/galaxy/pull/6239#issuecomment-393300753>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAnYCFminDGG8rO8_P_VnAYnVId1a9Zuks5t3vxZgaJpZM4URxpX>.

lparsons committed:
Set pid and log file for paster in server_args

This avoids setting log and pid file in paster_args so that with
GALAXY_RUN_ALL they can be set individually, but ensures they are set
when running paster otherwise.

lparsons commented:
@nsoranzo I believe this fixes the issue, at least it improves things. @natefoo Any comments?

nsoranzo commented:
This change will also fix an issue I just discovered, namely that `./run.sh restart` was using `paster.pid` and `paster.log` instead of `galaxy.pid` and `galaxy.log` , so:
```console
$ ./run.sh --daemon
$ ./run.sh restart
```
ends up with the first process not being killed and the second failing with `Address already in use`.

LGTM!

nsoranzo commented:
And also `./run.sh status` was broken in the same way:
```console
$ ./run.sh status
...
executing: python ./scripts/paster.py serve config/galaxy.ini  status
No PID file paster.pid
```

dannon commented:
@nsoranzo @lparsons Looks like the last change there changes may have inadvertently disabled the 'standard' non-logfile console running via run.sh, since --log-file is always passed now.

nsoranzo commented:
Sorry, I think I know how to fix this regression.

dannon commented:
@nsoranzo Ok, great!  Sorry I didn't notice this when testing the first time through.

lparsons commented:
Oops, sorry about that too. @nsoranzo did you already work up a patch? If not, I'd be happy to since I broke it..  ;-)

nsoranzo commented:
@lparsons Patch done, see https://github.com/galaxyproject/galaxy/pull/6262.
